include/zephyr/linker/linker-defs.h

```diff
@@ -294,17 +294,24 @@ extern char z_kobject_data_begin[];
 #ifdef CONFIG_THREAD_LOCAL_STORAGE
 extern char __tdata_start[];
 extern char __tdata_end[];
-extern char __tdata_size[];
 extern char __tdata_align[];
 extern char __tbss_start[];
 extern char __tbss_end[];
-extern char __tbss_size[];
 extern char __tbss_align[];
 extern char __tls_start[];
 extern char __tls_end[];
 extern char __tls_size[];
 #endif /* CONFIG_THREAD_LOCAL_STORAGE */
 
+#ifdef __riscv_cmodel_medany
```

Comment:
We have `CONFIG_RISCV_CMODEL_MEDANY`, maybe we would prefer to use Kconfig-derived defs here? https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_RISCV_CMODEL